libpod/image/image.go

```diff
@@ -149,43 +210,344 @@ func (i *Image) ID() string {
+		Os:              ociv1Img.OS,
+		ContainerConfig: &ociv1Img.Config,
+		Version:         info.DockerVersion,
+		Size:            size,
+		VirtualSize:     size + imgSize,
+		Annotations:     annotations,
+		Digest:          imgDigest,
+		Labels:          info.Labels,
+		RootFS: &inspect.RootFS{
+			Type:   ociv1Img.RootFS.Type,
+			Layers: ociv1Img.RootFS.DiffIDs,
+		},
+		GraphDriver: driver,
+	}
+	return data, nil
+}
+
+func annotations(manifest []byte, manifestType string) map[string]string {
+	annotations := make(map[string]string)
+	switch manifestType {
+	case ociv1.MediaTypeImageManifest:
+		var m ociv1.Manifest
+		if err := json.Unmarshal(manifest, &m); err == nil {
+			for k, v := range m.Annotations {
+				annotations[k] = v
+			}
+		}
+	}
+	return annotations
+}
+
+func (i *Image) toImageRef() (types.Image, error) {
+	ref, err := is.Transport.ParseStoreReference(i.imageruntime.store, "@"+i.ID())
+	if err != nil {
+		return nil, errors.Wrapf(err, "error parsing reference to image %q", i.ID())
+	}
+	imgRef, err := ref.NewImage(nil)
+	if err != nil {
+		return nil, errors.Wrapf(err, "error reading image %q", i.ID())
+	}
+	return imgRef, nil
+}
+
+// Decompose an Image
+func (i *Image) Decompose() error {
+	return types2.NotImplementedError
+}
+
+// TODO: Rework this method to not require an assembly of the fq name with transport
+/*
+// GetManifest tries to GET an images manifest, returns nil on success and err on failure
+func (i *Image) GetManifest() error {
+	pullRef, err := alltransports.ParseImageName(i.assembleFqNameTransport())
+	if err != nil {
+		return errors.Errorf("unable to parse '%s'", i.Names()[0])
+	}
+	imageSource, err := pullRef.NewImageSource(nil)
+	if err != nil {
+		return errors.Wrapf(err, "unable to create new image source")
+	}
+	_, _, err = imageSource.GetManifest(nil)
+	if err == nil {
+		return nil
+	}
+	return err
+}
+*/
+
+// GetImage retrieves an image matching the given name or hash from system
+// storage
+// If no matching image can be found, an error is returned
+func (ir *ImageRuntime) GetImage(image string) (*Image, error) {
+	var img *storage.Image
+	ref, err := is.Transport.ParseStoreReference(ir.store, image)
+	if err == nil {
+		img, err = is.Transport.GetStoreImage(ir.store, ref)
+	}
+	if err != nil {
+		img2, err2 := ir.store.Image(image)
+		if err2 != nil {
+			if ref == nil {
+				return nil, errors.Wrapf(err, "error parsing reference to image %q", image)
+			}
+			return nil, errors.Wrapf(err, "unable to locate image %q", image)
+		}
+		img = img2
+	}
+	newImage := ir.newFromStorage(img)
+	return newImage, nil
+}
+
+// GetImages retrieves all images present in storage
+// Filters can be provided which will determine which images are included in the
+// output. Multiple filters are handled by ANDing their output, so only images
+// matching all filters are included
+func (ir *ImageRuntime) GetImages(params *libpod.ImageFilterParams, filters ...libpod.ImageFilter) ([]*Image, error) {
```

Comment:
I want to say we should just drop the filtering here and keep things super simple. We can put it back in later if we need it.

Reply:
Or perhaps rename this GetFilteredImages() and then create a simpler GetImages()?

Reply:
Image filtering removed and simplified this function.